File: propertyResetterJSON.py
import json

# 026Level文件
file_path = "026NewLevel.level"
# 027Level文件
file_path2 = "NewLevel.level"
replace_data_dict = dict()


def main():
    with open(file_path, 'r', encoding='utf-8-sig') as f:
        data = json.load(f)
        scene_list = data["Scene"]
        for scene_item in scene_list:
            script = scene_item.get("Script")
            # 判空
            if script:
                script_component = script.get("ScriptComponent")
            # 去除空数据
            if script_component:
                # 遍历所有guid对应的json对象（key为guid）
                for k, v in script_component.items():
                    # 跳过包含“相机”的项
                    # if v.get("cameraMode"):
                    #     continue
                    replace_data_dict[k] = v
            # print(script_component)

        f.close()
    out_str = ""
    with open(file_path2, 'r+', encoding='utf-8-sig') as f:
        data = json.load(f)
        scene_list = data["Scene"]
        for scene_item in scene_list:
            script = scene_item.get("Script")
            # 判空
            if script:
                script_component = script.get("ScriptComponent")
            if script_component:
                # 遍历所有guid对应的json对象（key为guid）
                for k, v in script_component.items():
                    script_component[k] = replace_data_dict[k]
        out_str = json.dumps(data)
        f.close()

    with open(file_path2, 'w', encoding='utf-8') as f:
        f.write(out_str)
        f.close()


if __name__ == '__main__':
    main()

File: propertyResetterJSON.js
const fs = require('fs');

// 026Level文件
const file_path = "026NewLevel.level";
// 027Level文件
const file_path2 = "NewLevel.level";
const replace_data_dict = {};
const static_replace_data_dict = {};

function traverseJSON(obj) {
    if (typeof obj === 'object' && obj !== null) {
        if ("Script" in obj) {
            const script = obj["Script"];
            if (script) {
                const scriptComponent = script["ScriptComponent"];
                if (scriptComponent) {
                    for (const [k, v] of Object.entries(scriptComponent)) {
                        replace_data_dict[k] = v;
                    }
                }
                const staticScriptComponent = script["StaticScriptComponent"];
                if (staticScriptComponent) {
                    for (const [k, v] of Object.entries(staticScriptComponent)) {
                        static_replace_data_dict[k] = v;
                    }
                }
            }
        }
        for (const value of Object.values(obj)) {
            traverseJSON(value);
        }
    } else if (Array.isArray(obj)) {
        for (const item of obj) {
            traverseJSON(item);
        }
    }
}

function writeTraverseJSON(obj) {
    if (typeof obj === 'object' && obj !== null) {
        if ("Script" in obj) {
            const script = obj["Script"];
            if (script) {
                const scriptComponent = script["ScriptComponent"];
                if (scriptComponent) {
                    for (const [k, v] of Object.entries(scriptComponent)) {
                        scriptComponent[k] = replace_data_dict[k] || v;
                    }
                }
                const staticScriptComponent = script["StaticScriptComponent"];
                if (staticScriptComponent) {
                    for (const [k, v] of Object.entries(staticScriptComponent)) {
                        staticScriptComponent[k] = static_replace_data_dict[k] || v;
                    }
                }
            }
        }
        for (const value of Object.values(obj)) {
            writeTraverseJSON(value);
        }
    } else if (Array.isArray(obj)) {
        for (const item of obj) {
            writeTraverseJSON(item);
        }
    }
    return obj;
}

function main() {

    let str1 = fs.readFileSync(file_path, 'utf-8')
    // 去除BOM字符
    if (str1.charCodeAt(0) === 0xFEFF) str1 = str1.slice(1);
    const data = JSON.parse(str1);
    traverseJSON(data);

    let str2 = fs.readFileSync(file_path2, 'utf-8');
    // 去除BOM字符
    if (str2.charCodeAt(0) === 0xFEFF) str2 = str2.slice(1);
    let data2 = JSON.parse(str2);
    data2 = writeTraverseJSON(data2);

    const outStr = JSON.stringify(data2, null, 2);
    fs.writeFileSync(file_path2, outStr, 'utf-8');

    console.log("脚本修改完成！");
}

main();